### DISCUSSION TITLE
an error when compile lammps with deepmd package

### DISCUSSION THREAD

**Discussion by leiwang1989** [deepmodeling/deepmd-kit#281] [Q&A]
../pppm_dplr.cpp: In member function ‘virtual void LAMMPS_NS::PPPMDPLR::compute(int, int)’:
../pppm_dplr.cpp:89:5: error: ‘cg_peratom’ was not declared in this scope
     cg_peratom->ghost_notify();
     ^~~~~~~~~~
../pppm_dplr.cpp:89:5: note: suggested alternative: ‘eatom’
     cg_peratom->ghost_notify();
     ^~~~~~~~~~
     eatom
../pppm_dplr.cpp:130:3: error: ‘cg’ was not declared in this scope
   cg->reverse_comm(this,REVERSE_RHO);
   ^~
../pppm_dplr.cpp:130:3: note: suggested alternative: ‘vg’
   cg->reverse_comm(this,REVERSE_RHO);
   ^~
   vg
../pppm_dplr.cpp:150:7: error: ‘cg_peratom’ was not declared in this scope
       cg_peratom->forward_comm(this,FORWARD_AD_PERATOM);
       ^~~~~~~~~~
../pppm_dplr.cpp:150:7: note: suggested alternative: ‘eatom’
       cg_peratom->forward_comm(this,FORWARD_AD_PERATOM);
       ^~~~~~~~~~
       eatom
../pppm_dplr.cpp:152:7: error: ‘cg_peratom’ was not declared in this scope
       cg_peratom->forward_comm(this,FORWARD_IK_PERATOM);
       ^~~~~~~~~~

**Comment by njzjz** [comment]
Can you enable kspace package and/or try a newer version of lammps?

**Comment by leiwang1989** [comment]
I actually try them both, it does not work. My CPU is from AMD, will this be a problem for conda and this? seem unlikely.  Everything was fine, Java, Bazel, tensorflow C++, deeomd C++, until the last step. It seems the compiler cannot find the header file which declares those functions and variables.  I can also try with cmake to compile lammps with deepmd, but I am not familiar with it, it can take a while If I focus on it.

**Comment by leiwang1989** [comment]
mpicxx -g -O3  -DLAMMPS_GZIP -DLAMMPS_MEMALIGN=64   -std=c++11 -DHIGH_PREC   -I/home/zit.bam.de/lwang1/tensorflow_root/include -I/home/zit.bam.de/lwang1/tensorflow_root/include -I/home/zit.bam.de/lwang1/deepmd_root/include/deepmd   -std=c++11 -DHIGH_PREC -I/home/zit.bam.de/lwang1/tensorflow_root/include -I/home/zit.bam.de/lwang1/deepmd_root/include/deepmd  -std=c++11 -DHIGH_PREC   -I/home/zit.bam.de/lwang1/tensorflow_root/include -I/home/zit.bam.de/lwang1/tensorflow_root/include -I/home/zit.bam.de/lwang1/deepmd-kit_root/include/deepmd   -std=c++11 -DHIGH_PREC   -I/home/zit.bam.de/lwang1/tensorflow_root/include -I/home/zit.bam.de/lwang1/tensorflow_root/include -I/home/zit.bam.de/lwang1/deepmd-kit_root/include/deepmd   -std=c++11 -DHIGH_PREC   -I/home/zit.bam.de/lwang1/tensorflow_root/include -I/home/zit.bam.de/lwang1/tensorflow_root/include -I/home/zit.bam.de/lwang1/deepmd-kit_root/include/deepmd  -DMPICH_SKIP_MPICXX -DOMPI_SKIP_MPICXX=1    -c ../fix_nvt.cpp   



For your information
